### PR TITLE
Add natspec for AMM

### PR DESCRIPTION
**Pull request by charlesjhongc**
Add natspec comments for AMM.

### CODE REVIEW

File: contracts/interfaces/IAMMWrapper.sol [hunk 1]
```diff
@@ -3,11 +3,22 @@ pragma solidity >=0.7.0;
...
 interface IAMMWrapper is IStrategyBase {
-    // Operator events
-    event SetDefaultFeeFactor(uint16 newDefaultFeeFactor);
-    event SetFeeCollector(address newFeeCollector);
-
+    /// @notice Emitted after swap with AMM
+    /// @param source The tag of the contract where the order is filled
+    /// @param transactionHash The hash of the transaction structure
+    /// @param userAddr The address of taker
+    /// @param relayed The hash of the order structure
```
Comment (NIC619) [inline contracts/interfaces/IAMMWrapper.sol:13]: Indicate if the transaction is relayed by a relayer?

File: contracts/interfaces/IAMMWrapperWithPath.sol [hunk 2]
```diff
@@ -5,7 +5,15 @@ pragma abicoder v2;
 import "./IAMMWrapper.sol";
 import "../utils/AMMLibEIP712.sol";
 
+/// @title IAMMWrapper Interface
```
Comment (NIC619) [inline contracts/interfaces/IAMMWrapperWithPath.sol:8]:
```suggestion
/// @title IAMMWrapperWithPath Interface
```

File: contracts/interfaces/IAMMWrapper.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/interfaces/IAMMWrapper.sol]: Fixed in af7f2a6

File: contracts/interfaces/IAMMWrapperWithPath.sol (hunk 2 above)
Comment (charlesjhongc) [reply contracts/interfaces/IAMMWrapperWithPath.sol]: Fixed in af7f2a6

File: contracts/interfaces/IAMMWrapper.sol [hunk 3]
```diff
@@ -34,6 +44,19 @@ interface IAMMWrapper is IStrategyBase {
...
     }
 
+    /// @notice Trade with AMM
+    /// @param _makerAddress The AMM maker address
+    /// @param _fromAssetAddress The input assest of the trade
+    /// @param _toAssetAddress The output assest of the trade
+    /// @param _takerAssetAmount The input amount of the trade
+    /// @param _makerAssetAmount The output amount of the trade
+    /// @param _feeFactor The factor used to calculate fee, only listed relayer can set it effectively
+    /// @param _spender The address of the user
```
Comment (pilagod) [inline contracts/interfaces/IAMMWrapper.sol:54]: Shall we rename `_spender` to `_userAddr` to align with he naming in implementation?

Comment (charlesjhongc) [reply contracts/interfaces/IAMMWrapper.sol]: Goot catch, fixed in 326d081

**Review by pilagod — APPROVED** [review]

**Review by NIC619 — APPROVED** [review]

**Pull request merged**